### ISSUE TITLE
Consider using TelephonyManager to get cell towers

### ISSUE THREAD

**Issue by vubma** [SecUSo/privacy-friendly-wifi-manager#21]
As per #16 the user claims that you could use TelephonyManager API instead to read cell towers. If this works I think it could be a better solution since it wouldn't require location access. At least personally I prefer to only enable location when I want to use it and not keep it on all the time. It might also be more battery efficient.

**Comment by doctor-owl** [comment]
Agreed. This would be a huge improvement especially for those of us who want to save battery. I specifically use this app to have my wifi automatically turn on/off when I leave/return home to get the most out of my battery life, as I too often forget to do it manually. But keeping location on all the time kinda defeats the purpose

**Comment by Powew230v** [comment]
It would be best not to rely on location for privacy and battery savings.

**Comment by petermayer** [comment]
Thanks for the suggestion! We will investigate this option.

**Comment by Ikosihenagon** [comment]
Hi,
are there any news regarding the usage of cell tower information from the Telephony Manager instead of the Location Provider? Would be great if there would be one possible solution to implement this!

Thanks a lot!
